google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/impl/Formats.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage.impl
+
+import akka.http.scaladsl.model.{ContentType, ContentTypes}
+import akka.stream.alpakka.googlecloud.storage.GoogleCloudStorageModel._
+import spray.json.{DefaultJsonProtocol, JsValue, RootJsonReader}
+
+import scala.util.Try
+
+@akka.annotation.InternalApi
+object Formats extends DefaultJsonProtocol {
+
+  private final case class BucketListResultJson(
```

Comment:
I modeled fields in case classes that seemed relevant to me, is that good approach or should I include all fields that exist in google cloud storage models?